distance-method/distance_method.py

```diff
@@ -0,0 +1,981 @@
+    query_proteome_fp: string
+      filepath to query proteome
+    ref_fp: string
+      filepath to reference proteome
+    working_dir: string
+      working directory path
+    e_value: float, optional
+      the cutoff E-value for BLASTP results
+    threads: integer
+      number of threads to use for running BLASTP
+    debug: boolean
+      if True, run function in debug mode
+
+    Returns
+    -------
+    out_file_fp: string
+      filepath to tabular alignment file output by
+      BLASTP
+    """
+    db_file_fp = join(working_dir, "%s" % basename(ref_fp))
+    # build blast database
+    makeblastdb_command = ["makeblastdb",
+                           "-in", ref_fp,
+                           "-out", db_file_fp,
+                           "-dbtype", "prot"]
+    proc = subprocess.Popen(makeblastdb_command,
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE,
+                            close_fds=True)
+    proc.wait()
+    stdout, stderr = proc.communicate()
+    if (stderr and debug):
+        print "[DEBUG] %s\n" % stderr
+
+    # launch blast
+    out_file_fp = join(
+        working_dir, "%s.blast" % basename(splitext(query_proteome_fp)[0]))
+    blastp_command = ["blastp",
+                      "-db", db_file_fp,
+                      "-query", query_proteome_fp,
+                      "-evalue", str(e_value),
+                      "-num_threads", str(threads),
+                      "-outfmt", "6 std qcovs",
+                      "-task", "blastp",
+                      "-out", out_file_fp]
+    proc = subprocess.Popen(blastp_command,
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE,
+                            close_fds=True)
+    proc.wait()
+    stdout, stderr = proc.communicate()
+    if (stderr and debug):
+        print "[DEBUG] %s\n" % stderr
+
+    return out_file_fp
+
+
+def parse_blast(alignments_fp,
+                hits,
+                gene_map,
+                debug=False):
+    """ Parse BLASTp alignment file into a dictionary where the keys are the
+    queries and the values are all the reference sequences to which the query
+    mapped with E-value cutoff score.
+
+    Parameters
+    ----------
+    alignments_fp: string
+      filepath to tabular alignment file output by BLASTP
+    hits: dictionary
+      dictionary storing query (gene) names as keys and the best aligning
+      reference sequences as values (one alignment per reference sequence)
+    gene_map: dictionary
+      "two-way" dictionary storing gene names as keys and their pseudo
+      names as values, and vica versa
+    debug: boolean
+      if True, run function in debug mode
+    """
+    # read blastp results
+    with open(alignments_fp, 'U') as alignments_f:
+        for line in alignments_f:
+            if debug:
+                sys.stdout.write("[DEBUG] %s" % line)
+            line = line.split()
```

Comment:
Are these always 2 elements? If so you could: `query, ref = line.split()`. Note that if you get more elements this line is gonna break